src/index.js

```diff
@@ -1,14 +1,115 @@
-import { Controller } from 'stimulus'
-import SlimSelect from 'slim-select'
-
+import { Controller } from "stimulus";
+import SlimSelect from 'slim-select';
+import 'jsdig';
 export default class extends Controller {
   connect() {
-    this.slimselect = new SlimSelect({
-      select: this.element
-    })
+    this.slimselect = new SlimSelect(this._slimSelectOptions());
   }
 
   disconnect() {
     this.slimselect.destroy()
   }
+
+  _ajax({ url, urlParam, minimumLength, minimumLengthMessage, debounceMs, ...processAjaxResultConfig }) {
+    return this._debounce((search, callback) => {
+      if (search.length < minimumLength) {
+        callback(minimumLengthMessage);
+        return;
+      }
+
+      url.searchParams.set(urlParam, search);
+      fetch(url)
+        .then(function (response) {
+          return response.json()
+        })
+        .then(function (json) {
+          const data = this._processAjaxResult(json, processAjaxResultConfig);
+          callback(data);
+        }.bind(this))
+        .catch(function(_error) {
+          console.error(_error);
+          callback(false);
+        });
+    }, debounceMs);
+  }
+
+  _processAjaxResult(data, { valueColumn, textColumn, rootColumn }) {
+    let root = data;
+    if(rootColumn) root = data.dig(rootColumn);
+
+    return root.map(el => {
+      return {
+        value: el.dig(...valueColumn.split('.')),
+        text: el.dig(...textColumn.split('.'))
+      }
+    })
+  }
+
+  _slimSelectOptions() {
+    return Object.assign(
+      { select: this.element },
+      this._castOptions({ ...this.element.dataset }),
+      this._ajaxOptions()
+    )
+  }
+
+  _castOptions(data) {
+    return Object.keys(data).reduce((acc, current) => {
+      if (data[current] === "true") {
+        acc[current] = true;
+      } else if (data[current] === "false") {
+        acc[current] = false;
+      } else {
+        acc[current] = data[current];
+      }
+      return acc;
+    }, {})
+  }
```

Comment:
Can we use the Stimulus Values API to simplify this? https://stimulus.hotwire.dev/reference/values

Reply:
Whilst developing it it crossed my mind to use the Stimulus Values API. But I discarded that idea as it would have 2 downsides:

1. As maintainer of this package you need to keep updating the options list and keep it in sync with SlimSelect, lowering the maintainability of this package.
2. Developers need to use: `data-<optionKey>-value` in their html, whilst this is a regular pattern for experienced developers, it adds some (small) overhead for newcomers not familiar with Stimulus 2.x

Especially the first argument weighed heavy as the resulting code `this._castOptions` is small and easy to understand. So I chose maintainability in the long run over simplicity in the short term.